### PR TITLE
Upgrade Serverless Framework

### PR DESCRIPTION
**Pull request by nnti**
## Purpose

This change upgrades serverless framework from v2 to v3. 

The need for upgrading is because of the following reasons:
- Serverless v2 is not receiving updates; patches and updates are made in v3
- Serverless v2 has some dependencies on vulnerable packages
- New functionality to stay up to date with AWS functions
- Minimize plugin delays and hang times

#### Linked Issues to Close

https://qmacbis.atlassian.net/browse/OY2-17283?atlOrigin=eyJpIjoiZDhlNGU4N2Y0YjY3NDgxNTgxNDI0MGE4YTUxMTA1NmYiLCJwIjoiaiJ9

## Approach

- Serverless framework was updated from version 2 to version 3 in the top level package.json.  
- As many plugins in the sls plugin ecosystem made updates to accommodate v3, all sls plugins were updated to latest ( yarn upgrade pluginname --latest ).  No issues were found.
- Deployment and app functionality was verified to work as intended.

listed below are some features that have been deprecated in version 2 along with some variables resolution.
- You no longer need the tilde true to unecrypt an encrypted SSM param
  v2
  ssm:/configuration/${self:custom.stage}/iam/path~true
  v3
  ssm:/configuration/${self:custom.stage}/iam/path
- In v3, the variablesResolutionMode option can be removed as the new engine becomes the default.
  v2
  ${self:service.name}
  v3
  ${self:service}
- Bucket name should not contain uppercase characters. Ensure that your branch name is lowercase because that string is included in the bucket name string.
- Ensure all plugins have been upgraded to its latest.

Serverless Framework v3 now supports the standard "--verbose" flag to output more details.
That option can be particularly useful in CI/CD, for example to get a detailed history of the CloudFormation deployment

## Learning

https://www.serverless.com/framework/docs/guides/upgrading-v3

## Assorted Notes/Considerations

_List any other information that you think is important... a post-merge activity, someone to notify, etc._

#### Pull Request Creator Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [x] Someone has been assigned this PR.
- [x] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,21 +26,23 @@
...
+    "serverless-plugin-warmup": "^7.1.0",
     "serverless-s3-bucket-helper": "CMSgov/serverless-s3-bucket-helper#0.1.1",
-    "serverless-stack-termination-protection": "^1.0.4",
+    "serverless-stack-termination-protection": "^2.0.0",
     "typescript": "^4.0.5",
     "yargs": "^16.1.1"
   },
-  "dependencies": {},
+  "dependencies": {
+    "global": "^4.4.0"
```
Comment (mdial89f) [inline package.json:44]: Was this actually needed?  I'm surprised to see it added

Comment (nnti) [reply package.json:44]: I'm not sure. I can take it out and see if anything crashes

**Review by mdial89f — APPROVED** [review]

**Review by mdial89f — CHANGES_REQUESTED** [review]
Hey the code is good, but the PR write up itself needs to align to the PR Templates.  I'll ping you on Slack.

**Review by nnti — COMMENTED** [review]
PR template information has been filled

**Comment by berryd** [comment]
> Ensured all services in MACPro works as intended in version 3.

MACPro refers to the Appian product we support.  MACPRO is the suite of products we support including WMS/MMDL, SEA Tool, Appian, and the new products forked from the serverless quick start.  

> The new design:

What was fundamentally changed with the design?  The hope for this PR would be to document whatever challenges were encountered when fielding this work such that other teams could field the same sorts of changes within their respective product offerings using this as a guide.

**Comment by dwhitecl** [comment]
@nnti  can we add more details in the description to make this more easily consumed by product teams.
Reading this PR doesnt tell me why we're upgrading. Also, some things I'd spell out are technical details as to why the ssm syntax changed, how you'll need to make this change in each serverless service as applicable to your particular project.

**Comment by nnti** [comment]
> @nnti can we add more details in the description to make this more easily consumed by product teams. Reading this PR doesnt tell me why we're upgrading. Also, some things I'd spell out are technical details as to why the ssm syntax changed, how you'll need to make this change in each serverless service as applicable to your particular project.

**Comment by nnti** [comment]
> > @nnti can we add more details in the description to make this more easily consumed by product teams. Reading this PR doesnt tell me why we're upgrading. Also, some things I'd spell out are technical details as to why the ssm syntax changed, how you'll need to make this change in each serverless service as applicable to your particular project.

**Comment by nnti** [comment]
I've included the issues I encountered during the version update process. Please take a look at the updates made to the PR doc and let me know if I should include anything else. @dwhitecl @berryd

**Review by dwhitecl — APPROVED** [review]

**Review by mdial89f — APPROVED** [review]

**Comment by mdial89f** [comment]
> > 
> 
> @nnti you'll need to get @mdial89f 's enhancement request closed in order to squash and merge. I see @dwhitecl approved. I can't close the enhancement request for whatever reason.

thanks for the heads up @kenrickveit 
resolved

**Pull request merged**